app/controllers/plans_controller.rb

```diff
@@ -43,4 +43,21 @@ def search
       # TODO
     end
   end
+
+  def set_autofinger_subscription
+    owner = current_account
+    interest = Account.find_by_username(params[:id])
+    autofinger = owner.interests_in_others.find_or_create_by_interest(interest.id)
+    success = autofinger.update_attributes(priority: params[:priority])
+    if success
+      if params[:priority] == '0'
+        notice = 'User was removed from your autoread list.'
+      else
+        notice = "User is now on your autoread list with priority level of #{params[:priority]}."
+      end
+    else
+      notice = "Could not change autoread priority. Please try again."
```

Comment:
Checking for problems and telling the user is great and I am really glad we are doing that.  Would trying again actually fix it though?  My suggestion is to make it as useful as possible to the user, even if it doesn't give them an extra step.  Maybe just a note to email someone about it, maybe some extra logging so we can find the error later if it does come up.  IIRC `autofinger.errors` will get populated with any validation failures after `autofinger.update_attributes`